tools/pepsi/src/aliveness.rs

```diff
@@ -26,15 +26,25 @@ pub async fn aliveness(arguments: Arguments) -> anyhow::Result<()> {
         Arguments::Disable { naos } => (false, naos),
     };
 
-    let tasks = naos.into_iter().map(|nao_address| async move {
-        let nao = Nao::new(nao_address.ip);
-        nao.set_aliveness(enable)
-            .await
-            .with_context(|| format!("Failed to set aliveness on {nao_address}"))
+    let multi_progress = ProgressIndicator::new();
+
+    let tasks = naos.into_iter().map(|nao_address| {
+        let multi_progress = multi_progress.clone();
+        async move {
+            let progress = multi_progress.task(nao_address.to_string());
+            progress.set_message("Setting aliveness...");
+
+            let nao = Nao::new(nao_address.ip);
+
+            progress.finish_with(
+                nao.set_aliveness(enable)
+                    .await
+                    .with_context(|| format!("Failed to set aliveness on {nao_address}")),
+            );
+        }
     });
 
-    let results = join_all(tasks).await;
-    gather_results(results, "Failed to execute some aliveness setting tasks")?;
+    join_all(tasks).await;
```

Comment:
Use [FuturesUnordered](https://docs.rs/futures/latest/futures/stream/struct.FuturesUnordered.html) instead because the order does not matter probably?

For an example, look in the Communication server.

Reply:
Thanks, fixed.